Content.Shared/Utility/EntitySystemExtensions.cs

```diff
@@ -0,0 +1,85 @@
+﻿#nullable enable
+using System.Diagnostics.CodeAnalysis;
+using Content.Shared.Physics;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Map;
+using Robust.Shared.Maths;
+
+namespace Content.Shared.Utility
+{
+    public static class EntitySystemExtensions
+    {
+        public static IEntity? SpawnIfUnobstructed(
+            this IEntityManager entityManager,
+            string? prototypeName,
+            EntityCoordinates coordinates,
+            CollisionGroup collisionLayer,
+            Box2? box = null,
+            bool approximate = false)
+        {
+            var mapCoordinates = coordinates.ToMap(entityManager);
+            return entityManager.SpawnIfUnobstructed(prototypeName, mapCoordinates, collisionLayer, box, approximate);
+        }
+
+        public static IEntity? SpawnIfUnobstructed(
+            this IEntityManager entityManager,
+            string? prototypeName,
+            MapCoordinates coordinates,
+            CollisionGroup collisionLayer,
+            Box2? box = null,
+            bool approximate = false)
+        {
+            box ??= Box2.UnitCentered;
+
+            foreach (var entity in entityManager.GetEntitiesIntersecting(coordinates.MapId, box.Value, approximate))
+            {
+                if (!entity.TryGetComponent(out IPhysicsComponent? physics))
+                {
+                    continue;
+                }
+
+                if (!physics.Hard)
+                {
+                    continue;
+                }
+
+                if (collisionLayer == 0 || (physics.CollisionMask & (int) collisionLayer) == 0)
+                {
+                    continue;
+                }
+
+                return null;
+            }
+
+            return entityManager.SpawnEntity(prototypeName, coordinates);
```

Comment:
Should use GetCollidingEntities directly if you're getting the physics components.

Reply:
Done. Also the nullables are in, just not for tests as those are really annoying with nullable and very few use it. Should I enable it for the test as well?

Reply:
Done for the test as well.